### ISSUE TITLE
add setExecutor() method to allow use of user provided thread pool

### ISSUE THREAD

**Issue by aoberoi** [opentok/Opentok-Java-SDK#95]
following the discussion in #84, it became apparent that resource management concerns were not being handled well.

we should implement a `setExecutor()` method, similar to that of AsyncHttpClientConfig, so that user's can provide their own thread pool.

**Comment by Michael-Frank** [comment]
Hi @aoberoi 
any update on this feature?
I would really appreciate to haven an option to inject my own executor with a custom thread factory that only spawns deamon threads. 
Without this option there can allways occur some race condtions, where a new openTok instance ist created but spring (or any other webserver framework that uses this sdk) failes to call the close() method on openTok. This results in the non-deamon "Timer wheel" thread to be keept alive indefinetly, which prevents the application from shutting down properly.
This is a bummer in unittests as well as managed PaaS stacks that dynamically start and stop backend instances.

Long story short:  a client library  :
a) should not create its own thread pools
b) should not spawn threads that are non-deamon
as these will mess with the application lifecyle.

**Comment by SMadani** [comment]
A complete rewrite of the SDK is in the works, will be integrated in [Vonage/vonage-java-sdk](https://github.com/Vonage/vonage-java-sdk). Can't provide any timelines for GA but there's currently a branch with beta versions.
